azDevOps/azure/templates/steps/deploy-terraform-apply.yml

```diff
@@ -0,0 +1,49 @@
+parameters:
+  environment: ''
+  terraform_artefact_path: ''
+  # Parameters specific to terraform as a map
+  tf_job_input_params: ''
+  tf_job_output_params: ''
+
+steps:
+- bash: pwd; ls -la ; sudo chmod -R 777 ./ ; sudo ls -la; env
+  displayName: 'Environment'
+  workingDirectory: ${{ parameters.terraform_artefact_path }}
+
+- bash: terraform plan -input=false -out=tfplan
+  displayName: 'Terraform: plan'
+  workingDirectory: ${{ parameters.terraform_artefact_path }}
+  env:
+    # Authentication (this comes from the AzureDevOps library as secured variables)
+    ARM_SUBSCRIPTION_ID: $(azure_subscription_id)
+    ARM_CLIENT_ID: $(azure_client_id)
+    ARM_CLIENT_SECRET: "$(azure_client_secret)"
+    ARM_TENANT_ID: $(azure_tenant_id)
+    # Terraform 
+    TF_VAR_name_environment: ${{ parameters.environment }}
+    ${{ each input in parameters.tf_job_input_params }}:
+      TF_VAR_${{ input.key }}: ${{ input.value }}
+
+- bash: terraform apply tfplan
+  displayName: 'Terraform: apply'
```

Comment:
We should start considering split the plan from apply
I've noticed a few times the plan would demonstrate conflicts(deleting cluster for example) and we couldn't cancel the operation prior to the deletion\re-creation of the resource

Reply:
yes, let's put that in the backlog though to be done as soon as. but can stay as is for now?